libbeat/tests/compose/compose.go

```diff
@@ -0,0 +1,238 @@
+package compose
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/libcompose/docker"
+	"github.com/docker/libcompose/docker/ctx"
+	"github.com/docker/libcompose/project"
+	"github.com/docker/libcompose/project/options"
+)
+
+// docker-compose project wrapper
+type composeProject struct {
+	p    project.APIProject
+	file string
+}
+
+type serviceInfo struct {
+	Name    string
+	Running bool
+	Healthy bool
+	// Has been up for more than 1 minute
+	Old bool
+}
+
+// EnsureUp starts all the requested services (must be defined in docker-compose.yml)
+// It returns a tearDown method that must be called once finished, to kill the containers
+func EnsureUp(t *testing.T, services ...string) func() {
+	compose, err := getComposeProject()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, service := range services {
+		err := compose.Start(service)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Kill no longer used containers
+	compose.KillOld(services)
+
+	for _, service := range services {
+		err := compose.Wait(service)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Return down function
+	return func() {}
```

Comment:
Is this empty on purpose? Should we at least log something here?

Reply:
In previous iterations I did a kill after every test, long story short: `go test` forkss its process by package to run tests, so it was a concurrency mess. Current solution uses the docker-compose lock file and delayed teardown, I will get rid of the `defer` hack.